base/math.jl

```diff
@@ -83,10 +83,20 @@ end
 
 
 """
-    evalpoly(x, p::Tuple)
+    evalpoly(x, p)
+
 Evaluate the polynomial ``\\sum_k p[k] x^{k-1}`` for the coefficients `p[1]`, `p[2]`, ...;
-that is, the coefficients are given in ascending order by power of `x`. This function
-generates efficient code using Horner's method with loops unrolled at compile time.
+that is, the coefficients are given in ascending order by power of `x`.
+Loops are unrolled at compile time if the number of coefficients is statically known, i.e.
+when `p` is a `Tuple`.
+This function generates efficient code using Horner's method if `x` is real, or using
+a Goertzel-like algorithm if `x` is complex.
```

Comment:
Maybe make this a footnote
```suggestion
a Goertzel-like [^DK62] algorithm if `x` is complex.

[^DK62]: Donald Knuth, Art of Computer Programming, Volume 2: Seminumerical Algorithms, Sec. 4.6.4.
```

Reply:
Whoa, TIL. Cool!